### PR TITLE
Added optionalDeviceInformation to IndividualEnrollment

### PR DESCRIPTION
**Pull request by dylanbulfinMS**
## Checklist
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-iot-sdk-csharp/blob/main/.github/CONTRIBUTING.md).
- [ ] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- [ ] This pull-request is submitted against the `main` branch.

### CODE REVIEW

File: provisioning/service/src/Config/IndividualEnrollment.cs [hunk 1]
```diff
@@ -212,6 +212,12 @@ public override string ToString()
         [JsonProperty(PropertyName = "registrationState", DefaultValueHandling = DefaultValueHandling.Ignore)]
         public DeviceRegistrationState RegistrationState { get; private set; }
 
+        /// <summary>
+        /// Optional Device Information.
```
Comment (drwill-ms) [inline provisioning/service/src/Config/IndividualEnrollment.cs:216]: https://github.com/Azure/azure-iot-sdk-csharp/pull/2297#discussion_r797218904

File: provisioning/service/tests/Config/IndividualEnrollmentTests.cs [hunk 2]
```diff
@@ -266,6 +270,7 @@ public void IndividualEnrollmentConstructorWithoutCapabilitiesJSONSucceed()
             Assert.IsTrue(individualEnrollment.Attestation is X509Attestation);
             Assert.AreEqual(SampleDeviceId, individualEnrollment.DeviceId);
             Assert.AreEqual(SampleIotHubHostName, individualEnrollment.IotHubHostName);
+            Assert.IsNotNull(individualEnrollment.OptionalDeviceInformation);
```
Comment (drwill-ms) [inline provisioning/service/tests/Config/IndividualEnrollmentTests.cs:273]: Shall we confirm the contents of the payload as well?

Comment (dylanbulfinMS) [reply provisioning/service/tests/Config/IndividualEnrollmentTests.cs:275]: Done

**Comment by abhipsaMisra** [comment]
There isn't any work being carried out on this at the moment. We'll revisit this in the future.

**Pull request closed without merging**